red_queen/cli.py

```diff
@@ -0,0 +1,185 @@
+)
+@click.option(
+    "-t",
+    "--benchmarkType",
+    "benchmarkType",
+    multiple=True,
+    type=click.Choice(benchmark_types),
+    help="enter the type of benchmark(s) here",
+)
+@click.option(
+    "-b",
+    "--benchmark",
+    "benchmark",
+    multiple=True,
+    type=click.Choice(benchmarks),
+    help="enter the specfic benchmark(s) here",
+)
+def main(compiler=None, benchmarkType=None, benchmark=None):
+    benchmark_paths = []
+    pytest_paths = ""
+    mydict = {}
+    m_tag = ""
+    if len(compiler) > 0:
+        m_tag = "-m"
+        compiler = compiler[0]
+    else:
+        compiler = ""
+    # ### This for loop ensures that we are able to run various benchmark types
+    i = 0
+    j = 0
+    ### Has a benchmark type been specified?
+    if len(benchmarkType) > 0:
+        # click.echo("passed test 0")
+        while i < len(benchmarkType):
+            ### Is the inputted benchmark type valid?
+            if set(benchmarkType).issubset(benchmark_category):
+                # click.echo("passed test 1")
+                ### Has a benchmark been specified?
+                if len(benchmark) > 0:
+                    # click.echo("passed test 2")
+                    ### Are the inputted benchmark(s) valid?
+                    if set(benchmark).issubset(benchmarks):
+                        # click.echo("passed test 3")
+                        ### Is the inputted benchmark within the inputted benchmark type suite?
+                        if set(benchmark).issubset(set(benchmark_category[benchmarkType[i]])):
+                            # click.echo("passed test 4")
+                            for j, _ in enumerate(benchmark):
+                                benchmark_paths.append(
+                                    benchmark_category[benchmarkType[0]][benchmark[j]]
+                                )
+                            pytest_paths = benchmark_paths
+                            run_benchmarks(pytest_paths, m_tag, compiler)
+                            # show_result()
+                            i += 1
+                else:
+                    mydict = benchmark_category[benchmarkType[0]]
+                    for v in mydict.values():
+                        benchmark_paths.append(v)
+                    pytest_paths = " ".join(tuple(benchmark_paths))
+                    run_benchmarks(pytest_paths, m_tag, compiler)
+                    show_result()
+                    i += 1
+    else:
+        question = input(f"Would you like to run all {len(benchmarks)} available benchmarks (y/n) ")
+        if question.lower() == "y":
+            for benchmark_list in benchmark_category.values():
+                for paths in benchmark_list.values():
+                    benchmark_paths.append(paths)
+            pytest_paths = " ".join(tuple(benchmark_paths))
+            run_benchmarks(pytest_paths, m_tag, compiler)
+            show_result()
+
+
+if __name__ == "__main__":
+    # main()
```

Comment:
Should this be commented? I assume if you call this file standalone you want it to run the CLI too:

```suggestion
    main()
```

Reply:
Good catch! I was testing out code and forgot comment back in `main()`.